vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/database/blackhole/BlackholeConnection.kt

```diff
@@ -0,0 +1,62 @@
+package org.vitrivr.engine.core.database.blackhole
+
+import io.github.oshai.kotlinlogging.KLogger
+import io.github.oshai.kotlinlogging.KotlinLogging.logger
+import org.vitrivr.engine.core.database.AbstractConnection
+import org.vitrivr.engine.core.database.ConnectionProvider
+import org.vitrivr.engine.core.database.blackhole.descriptors.BlackholeDescriptorInitializer
+import org.vitrivr.engine.core.database.blackhole.descriptors.BlackholeDescriptorReader
+import org.vitrivr.engine.core.database.blackhole.descriptors.BlackholeDescriptorWriter
+import org.vitrivr.engine.core.database.blackhole.retrievable.BlackholeRetrievableInitializer
+import org.vitrivr.engine.core.database.blackhole.retrievable.BlackholeRetrievableReader
+import org.vitrivr.engine.core.database.blackhole.retrievable.BlackholeRetrievableWriter
+import org.vitrivr.engine.core.database.descriptor.DescriptorReader
+import org.vitrivr.engine.core.database.retrievable.RetrievableInitializer
+import org.vitrivr.engine.core.database.retrievable.RetrievableReader
+import org.vitrivr.engine.core.database.retrievable.RetrievableWriter
+import org.vitrivr.engine.core.model.descriptor.Descriptor
+import org.vitrivr.engine.core.model.metamodel.Schema
+
+
+/** Defines [KLogger] of the class. */
+internal val LOGGER: KLogger = logger("org.vitrivr.engine.database.blackhole.BlackholeConnection")
```

Comment:
Why is the 'logger of the class' not a field of the class?

Reply:
This is an idiomatic way of generating loggers using `oshai/kotlin-logging`. According to the [official documentation](https://github.com/oshai/kotlin-logging?tab=readme-ov-file) "behind the scenes `val logger` will be created in the class, with the class/file name"